Compiler/src/utilities.jl

```diff
@@ -329,7 +329,7 @@ end
 
 inlining_enabled() = (JLOptions().can_inline == 1)
 
-function coverage_enabled(m::Module)
+function instrumentation_enabled(m::Module, only_if_affects_optimizer::Bool)
```

Comment:
The name here sounds like it means "check whether instrumentation is enabled for `m`" but really it seems to determine whether `m` needs to have instrumentation. Maybe `requires_instrumentation`?

Reply:
"should I turn `x` on" vs "is `x` turned on". This doesn't use `m` in any way other than to check if the root is `Core` or `Base`, otherwise it's just examining `JLOptions` which is a system property, not a module property.

But this is obviously a minor point, if you disagree then feel free to ignore this.

Reply:
I guess I don't disagree, I just didn't linguistically see the difference, since that seems like the difference between specifying whether this is the policy and mechanism, but this is effectively both of them